src/internals/configuration.rs

```diff
@@ -0,0 +1,66 @@
+use std::alloc::{alloc, Layout};
+use std::ffi::c_void;
+
+use yara_sys;
+
+use crate::errors::*;
+
+/// A `ConfigName` is enum of available configuration options
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ConfigName {
+    /// Stack size to use for YR_CONFIG_STACK_SIZE
+    StackSize,
+    /// Maximum number of strings to allow per yara rule. Will be mapped to YR_CONFIG_MAX_STRINGS_PER_RULE
+    MaxStringsPerRule,
+    /// Maximum number of bytes to allow per yara match. Will be mapped to YR_CONFIG_MAX_MATCH_DATA
+    MaxMatchData,
+}
+
+impl ConfigName {
+    #[cfg(unix)]
+    pub fn to_yara(&self) -> u32 {
+        use self::ConfigName::*;
+        let res = match self {
+            StackSize => yara_sys::_YR_CONFIG_NAME_YR_CONFIG_STACK_SIZE,
+            MaxStringsPerRule => yara_sys::_YR_CONFIG_NAME_YR_CONFIG_MAX_STRINGS_PER_RULE,
+            MaxMatchData => yara_sys::_YR_CONFIG_NAME_YR_CONFIG_MAX_MATCH_DATA,
+        };
+        res as u32
+    }
+
+    #[cfg(windows)]
+    pub fn to_yara(&self) -> i32 {
```

Comment:
Why is the type different for the two platforms?
Maybe you can just do

```rust
#[cfg(unix)]
use RawType = u32;
#[cfg(windows)]
use RawType = i32;

pub fn to_yara(&self) -> RawType { ... }
```

In addition, we might miss other platforms.

Reply:
Thank you, fix!

Reply:
More read about this problem here: https://github.com/rust-lang/rust-bindgen/issues/1096